_projects/automate_vulture_whitelist.md

```diff
@@ -0,0 +1,84 @@
+---
+name: "Automate Whitelist creation for vulture"
+collaborating_projects:
+  - coala
+  - vulture
+desc: "Automatically create whitelists and detect false positives for vulture."
+developers_involved:
+  - RJ722
+mentors:
+  - AbdealiJK
+  - jendrikseipp
+  - The-Compiler
+difficulty: medium
+status: in_progress
+initiatives:
+  - GSoC
+issues:
+- "https://github.com/jendrikseipp/vulture/issues/109"
+- "https://github.com/jendrikseipp/vulture/issues/8"
+markdown: automate_vulture_whitelist.md
+requirements:
+  - "Familiarity with vulture and how whitelists work."
+tags:
+  - Vulture
+  - AST
+  - coverage
+---
+
+The main aim of this project is to automatically create whitelists (to be
+utilized by vulture) for projects projects using
+[`sip`](https://www.riverbankcomputing.com/software/sip/intro) and
+[`shiboken`](http://code.qt.io/cgit/pyside/pyside-setup.git/tree/sources/shiboken2)
+to generate Python bindings for C++ code, most significant of them being PyQt.
+Once implemented, a whitelist generated for PyQt (let's say) through this
+generator could then be universally utilised across all PyQt projects.
+
+#### IDEAS
+
+* Create a command line tool which takes in `sip` files as input and spits a
+  whitelist with all the virtual functions defined in the `sip` files. (Add
+  Documentation and tests)
+* Use this tool to generate a whitelist for PyQt and ship it with vulture. This
+  will reduce overhead for PyQt projects which use vulture.
+* Allow passing coverage.xml to vulture via `--make-whitelist` and let it find
+  the list of false-positive functions. (Add Documentation and tests as well)
+* Write a whitelist generator for shiboken, the binding generator used for
+  PySide2.
+
+#### MILESTONES
+
+##### COMMUNITY BONDING
+* Discussion on whether to implement a new Tracer or to directly consume `XML`
+  output from `coverage.py`.
+* Get familiar with `coverage.xml`.
+* Take up small tasks from
+  [TODO](https://github.com/jendrikseipp/vulture/tree/master/TODO.rst).
+
+##### CODING PHASE 1
+* The whitelist generator should be completely implemented.
```

Comment:
As mentioned at https://github.com/coala/projects/pull/571#pullrequestreview-119643407 , it would be good if it was designed foremost a re-usable library which did the hard work with a decent API, and then a CLI interface which did the 'vulture' specific whitelist generation.

Reply:
I see nothing in the project brief about this, and no direct reply here.  Please reply directly on the PR diff to keep the conversion together.  https://github.com/coala/projects/pull/571#issuecomment-392283437 seems to be a reply, where you tell me 'no'.
Why do you want to write a tool which can not be used by other people?

Reply:
Hehe @jayvdb! I didn't say that no one would use it - I'm just saying that it should be user ended rather than being developed with an API.

Here, in this case, we are trying to develop a CLI app to create whitelists for vulture (via looking up for virtual functions in the XML output) and parsing other information from XML and presenting it is not something to look up to.

Reply:
Only the output format is vulture specific.  That is the easy part.  The difficult part is all of the parsing, and that is usable by coala, and you are refusing to build it with functions that coala can use.

Reply:
Im not sure what the problem is with exposing an API, but assuming the code is half decently designed, we can add it later, so closing this off for now.